app/src/main/java/com/medmapper/v33001/MainViewModel.kt

```diff
@@ -14,42 +15,31 @@ import com.medmapper.v33001.dto.Medicine
 import com.medmapper.v33001.dto.User
 import com.medmapper.v33001.service.IMedicineService
 
-class MainViewModel(var medicineService : IMedicineService) /*= MedicineService())*/ : ViewModel() {
+class MainViewModel(var medicineService : IMedicineService) : ViewModel() {
 
-    internal val NEW_MEDICATION = "New Medication"
-    var medicine : MutableLiveData<List<Medicine>> = MutableLiveData<List<Medicine>>()
+    private val NEW_MEDICATION = "New Medication"
+    private val _medicine = MutableLiveData<List<Medicine>>()
+    val medicine: LiveData<List<Medicine>> get() = _medicine
     var selectedMedicine by mutableStateOf(Medicine())
     var user: User? = null
 
-    private lateinit var firestore: FirebaseFirestore
-
-    init {
-        firestore = FirebaseFirestore.getInstance()
-        firestore.firestoreSettings = FirebaseFirestoreSettings.Builder().build()
+    private val firestore: FirebaseFirestore = FirebaseFirestore.getInstance().apply {
+        firestoreSettings = FirebaseFirestoreSettings.Builder().build()
```

Comment:
- The medicine property is mutable and is exposed as a public field. It can be made immutable by using the "val" keyword and making it private. We can expose it as a read-only property by providing a public getter method.

    Source: https://kotlinlang.org/docs/properties.html#mutable-properties-and-variables 
    Source: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/map-not-null.html

Reply:
Encapsulation is often a good idea.